src/HttpRequest.h

```diff
@@ -249,6 +249,9 @@ class HttpRequest: public Http::Message
 
     virtual void configureContentLengthInterpreter(Http::ContentLengthInterpreter &) {}
 
+    /// checks whether the Content-Length header (or lack of) is acceptible
+    bool canUseContentLength() const;
```

Comment:
Except, this method is not about whether the body is permissible. It is about whether the state of Content-Length (either present or absent) will cause a framing error. It is only relevant to to traffic relying on that Content-Length state for framing.

There is a rather large and complex amount of logic in client_side.cc following the call to this method which goes deep into the payload contents, format, syntax etc which would be relevant to a generic "checkBody()". Refactoring all of that is not in scope here.

Reply:
> Except, this method is not about whether the body is permissible

I think it is or should be, but if you can suggest a better name/description, please do. The current ones are not going to work for the reasons I have stated in this change request. Attacking suggested fixes alone does not help with making progress. If you do not like those suggestions, please propose a better fix! I certainly do not insist on any specific name or wording.

> It is about whether the state of Content-Length (either present or absent) will cause a framing error.

Agreed. The name does not match this scope well (as I already noted). Also, the method currently does not cover all Content-Length cases, but after we move the `chunked` case inside the method, it will. At that point, the current method name will become even less appropriate, of course.

> There is a rather large and complex amount of logic in client_side.cc following the call to this method which goes deep into the payload contents, format, syntax etc  ... Refactoring all of that is not in scope here.

No such refactoring has been requested. This change request is about the being-adjusted method name, parameter, and description. The matching caller changes are very small, and the PR already changes the caller and the method, so we are not talking about some serious new refactoring here. If you prefer, I can make these changes myself.

Reply:
> > Except, this method is not about whether the body is permissible
> 
> I think it is or should be, 

See the other thread about moving chunking tests. The decision about body permissiveness is made by the caller.

Reply:
You don't consider a const method with side effects being a "smell" ?

Reply:
I do not like the name checkBody as it implies that we are checking the content of the body.

checkEntityFraming() seems more appropriate when C-L and T-E are both being validated. If you are okay with that name I will make that change and move the chunking check inside.

Reply:
I have no strong objections against checkEntityFraming(). The "entity" part sounds unnecessary complex and possibly technically wrong to me -- I would prefer the simpler "body" -- but I can accept "entity" to make progress. Please proceed with any variation of those names.

Reply:
Done in 56a86ec